### ISSUE TITLE
Integrate with DocumentManager/FileUploader

### ISSUE THREAD

**Issue by jlsjonas** [laravel-enso/FormBuilder#41]
This is  a **feature request**.

### Prerequisites
* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you check the documentation?
* [x] Did you perform a cursory search?

### Description

Allow adding a "document" type to the form builder; integrating with (parts of?) DocumentManager//fileUploader to allow easily saving a specific document/image along with a form/model

**Comment by jlsjonas** [comment]
I'm assuming this is currently not a _priority_ right now?

Could you point me in the right direction to implement this in the framework (places/sub-projects that will definitely have to be modified to properly support this)
I'll look into making a PR for it in that case (we need to add images to a form)

Alternatively, if the framework regarding images/documents is changing soon, I'll resort to a custom object instead.

**Comment by aocneanu** [comment]
This is a very nice feature and I want it added as soon as possible. The reason it was left behind is because we didn't had the time to really think it through.

Let's use this issue to discuss what the expectations are.

 @gandesc  please add your suggestions here.

From the FE point of view it can be a nice file field with and upload button with its presence controlled from the json template. It should have a `multiple` option.

But once we upload one or more files, what do we expect from this field to be visual represented in the edit form? How will one file/more files be displayed?

About behaviour:
- do we want additional controls, like the ability to delete file(s) from the field?
- at what moment the file(s) should be persisted? Once the upload is done? Or when we click the `Store`/`Update` actions?

From the BE point of view, what are the implications regarding the current `FileManager`, and so on.

**Comment by jlsjonas** [comment]
I think the best strategy from the FE perspective is to include a `"content"`-like option, or create 2 `type` fields to differentiate between images & files.

## Images

- use a thumbnail-like grid as the upload button/dropzone
- (preview of) images _replaces_ the thumbnail visually
  - multiple: the upload zone moves to the right. (`prepend` the image instead)
- the top-right of (each) image preview should contain an `x` to remove the image from the upload queue

## Files

Similar to images, but using a list instead of thumbnails, and I would keep the upload zone _row_ fixed at the top instead.
each _preview_ (row) should ideally contain:
- an icon indicating the file type
- the file name, cropping _in between_ if the width is insufficient (i.e.: `con...456` vs. `concep...`)
- an `x` to remove the file from the queue

## Behaviour

(in my opinion, of course)
- Deleting files should definitely be possible (in a later stage perhaps allowing an alias?)
- persist should happen when clicking `Save`/`Update`, as would be expected from a form (in a later stage perhaps provide the **option** to upload the file to a temp folder to increase fluidity; however this should remain an option

**Comment by aocneanu** [comment]
>I think the best strategy from the FE perspective is to include a "content"-like option, or create 2 type fields to differentiate between images & files.

sounds good.

>EDIT: a specific max filesize / filetype option would also be recommended

filesize seems ok, but for filetypes we will stick with the above (content) and do extra validations on the back-end

>Images
use a thumbnail-like grid as the upload button/dropzone
(preview of) images replaces the thumbnail visually
multiple: the upload zone moves to the right. (prepend the image instead)
the top-right of (each) image preview should contain an x to remove the image from the upload queue
Files
Similar to images, but using a list instead of thumbnails, and I would keep the upload zone row fixed at the top instead.
each preview (row) should ideally contain:

sound good.

>an icon indicating the file type
the file name, cropping in between if the width is insufficient (i.e.: con...456 vs. concep...)
an x to remove the file from the queue
Behaviour
(in my opinion, of course)

can be a future development

It would be cool if you could find/post some pics as examples, maybe found nice implementations elsewhere.

**Comment by jlsjonas** [comment]
>filesize seems ok, but for filetypes we will stick with the above (content) and do extra validations on the back-end

Sounds good (though for "files" it might be really useful (limit to .pdf, .csv, ...) )

>can be a future development

 👍 

# Inspiration

## Images

my original thought (quick mockup):
pre-upload:
![image](https://user-images.githubusercontent.com/932193/47087415-3aaaf600-d21c-11e8-950e-f7b893901b61.png)

after a couple uploads:
![image](https://user-images.githubusercontent.com/932193/47087448-4ac2d580-d21c-11e8-8dfb-3d0aa8643ca9.png)

extra inspiration (dropzonejs):
[![](http://cdn.instantshift.com/wp-content/uploads/2015/10/free-html5-css3-jquery-upload-file-form-scripts-03.jpg)](http://www.dropzonejs.com/)


## files

some combination of these (with a clean _enso-look_)

![](http://cdn.instantshift.com/wp-content/uploads/2015/10/free-html5-css3-jquery-upload-file-form-scripts-01.jpg)

![](http://cdn.instantshift.com/wp-content/uploads/2015/10/free-html5-css3-jquery-upload-file-form-scripts-17.jpg)

![](http://cdn.instantshift.com/wp-content/uploads/2015/10/free-html5-css3-jquery-upload-file-form-scripts-28.jpg)

**Comment by jlsjonas** [comment]
Also note that --ideally--  clicking on the image _preview_ shows a bigger (modal?) view of it

use case: form edit being used by another user than the one creating the form; wanting to inspect the uploaded pictures

**Comment by aocneanu** [comment]
apparently makes more sense to build a custom wrapper around https://www.dropzonejs.com/

**Comment by jlsjonas** [comment]
vue-dropzone (linked above) looks like a better fit though; it's already in Vue & has closer feature-parity to the implementation features above (deleting; slots for styling; already a vue component; ...)
but I'll let you decide what you seem more fit. (if I compare the docs I'm pretty sure vue-dropzone includes all options that make sense as props/similar; so either should work)

**Comment by jlsjonas** [comment]
I've noticed that Dropzone got added last week; is this feature (partially) implemented that are just missing docs? or is it just preparations that had been made?
I can definitely help with testing/finalizing if useful.

**Comment by jlsjonas** [comment]
Ok, no problem; was just curious.

I've been noticing the great work being pushed recently ;)

**Comment by curtisdelicata** [comment]
Is this still planned?

**Comment by mauthi** [comment]
Hi @aocneanu,

I just started to create the same issue but luckily I found this existing one.

Can you please give me a quick update for this feature? I would need it for my project too.
Thx.

**Comment by mauthi** [comment]
Hi @aocneanu,

Hope you are doing well?

Do you maybe have any update for this if this is still planed?

**Comment by robbykrlos** [comment]
What a roller-coaster of great ideas guys. Was this feature finally made? It's Enso 7.0.0 now.

**Comment by aocneanu** [comment]
Thank you for your contribution.

As part of the Laravel Enso 9.0.0 release reset, we are closing the current backlog of GitHub issues and pull requests created before April 23, 2026.

Enso 9.0.0 is one of the largest frontend/backend releases in the history of Laravel Enso, bringing the ecosystem up to date with Laravel 13, PHP 8.3+, Vue 3.5, Vite 7, Pinia, a redesigned active Enso UI layer, and refreshed public documentation.

Because of the scale of these changes, older reports and pull requests may no longer apply cleanly to the Enso 9 line. If this issue or pull request is still relevant on top of the current Enso 9 stack, please open a new issue or pull request with updated context.

Thank you for understanding.
